spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -283,7 +283,8 @@ public Statistics estimateStatistics() {
       return new Stats(0L, 0L);
     }
 
-    if (filterExpressions == null || filterExpressions == Expressions.alwaysTrue()) {
+    // estimate stats using snapshot summary only for partitioned tables (metadata tables are unpartitioned)
+    if (table.spec().fields().size() > 0 && filterExpression() == Expressions.alwaysTrue()) {
```

Comment:
I recently added `isUnpartitioned` since we do this check fairly often.

Reply:
I try to avoid negation in conditions to keep them easy to read. It looks OK here, updated.